test/dynamo/test_install_params_as_graph_attr.py

```diff
@@ -0,0 +1,156 @@
+# Owner(s): ["module: dynamo"]
+from typing import Callable
+
+import torch
+import torch._dynamo
+import torch._dynamo.test_case
+import torch._dynamo.testing
+from torch._dynamo.testing import EagerAndRecordGraphs
+from torch.fx.graph_module import GraphModule
+
+
+def compile_and_extract_graph(
+    fn, *args, **kwargs
+) -> tuple[Callable, list[torch.fx.GraphModule]]:
+    backend = EagerAndRecordGraphs()
+    result_fn = torch.compile(backend=backend)(fn)
+    # Run fn to capture graph
+    _ = result_fn(*args, **kwargs)
+    return result_fn, backend.graphs
+
+
+def get_num_input_nodes(graph: GraphModule) -> int:
+    """Returns the number of input nodes in the input GraphModule
+
+    NOTE: Assumes a bijection between input nodes and placeholders
+    but, this is not always the case as symbolic tracing can introduce
+    additional placeholders.
+
+    For these test cases, there is no symbolic tracing, so this is
+    a safe assumption
+    """
+    placeholder_cnt = 0
+    for node in graph.graph.nodes:
+        if node.op == "placeholder":
+            placeholder_cnt += 1
```

Comment:
One other thing you could do to ensure that you get tensor inputs is

```
if node.op == "placeholder" and isinstance(node.meta["example_value"], torch.Tensor):
```

This will remove symints.